src/unionfind.rs

```diff
@@ -3,12 +3,14 @@
 //!
 //! This implementation uses interior mutability for `find`.
 use crate::util::HashMap;
-use crate::{Id, Symbol, Value};
+use crate::{Symbol, Value};
 
 use std::cell::Cell;
 use std::fmt::Debug;
 use std::mem;
 
+pub type Id = u64;
```

Comment:
Motivation was to take code that looked like `value.bits = u64::from(egraph.unionfind.find(Id::from(value.bits as usize)))` and change it into `value.bits = egraph.unionfind.find(value.bits)`

Reply:
I would prefer to keep the old struct wrapper since it makes our code less error-prone. We can implement `Into<usize>` and alike to make this less tedious if we have not already. Happy to be convinced though.

On the other hand, it does seem to be a problem that `value.bits` uses u64 while `Id` uses `usize`.

Reply:
`Into` was implemented before, so not all of the code was as bad as above. Some of it looked like `value.bits = egraph.unionfind.find((value.bits as usize).into()).into()`, which is better but still unnecessary IMO.

Reply:
Conclusion: both ways are fine, `struct Id` is slightly safer, `type Id` is less verbose, Yihong prefers the safety while I prefer the simplicity